### PR TITLE
Expand NPC config table and sprite group table

### PR DESCRIPTION
**Pull request by tolmar**
As discussed in the discord. This lets NPC configs and sprite groups expand.

Please test and carefully review, I could easily have missed something up on integration.

Adding new NPCs/sprites crashes EbProjEditor. I'll be making another patch for that repo.

### CODE REVIEW

File: coilsnake/modules/eb/SpriteGroupModule.py [hunk 1]
```diff
@@ -21,6 +21,13 @@ class SpriteGroupModule(EbModule):
                    (0x130000, 0x13ffff),
                    (0x140000, 0x14ffff),
                    (0x150000, 0x154fff)]
+    SPRITE_GROUP_TABLE_RELOCATIONS = [
+        0x001DF9,
+        0x001E79,
+        0x001FE0,
+        0x007A8B,
+        0x04B1D0,
+    ]
```
Comment (charasyn) [inline coilsnake/modules/eb/SpriteGroupModule.py:30]: Can you change these to `AsmPointerReference` instances instead? I think that would be nicer than needing to know they're all AsmPointers

Additionally, can you rename the attribute to `SPRITE_GROUP_TABLE_REFERENCES` to match with the new naming convention?

File: coilsnake/modules/eb/ExpandedTablesModule.py [hunk 2]
```diff
@@ -13,18 +12,46 @@ def write(self, rom, address):
...
+        self.offset = offset
+
+    def write(self, rom, address):
+        log.info("Writing xl pointer at " + hex(self.offset))
+        write_xl_pointer(rom, self.offset, address)
 
 log = logging.getLogger(__name__)
 
 
+
```
Comment (charasyn) [inline coilsnake/modules/eb/ExpandedTablesModule.py:26]: nit: remove newly added newline

File: coilsnake/modules/eb/ExpandedTablesModule.py [hunk 3]
```diff
@@ -2,8 +2,7 @@
 from coilsnake.model.eb.table import eb_table_from_offset
 from coilsnake.modules.eb.EbModule import EbModule
 from coilsnake.util.common.yml import yml_load
-from coilsnake.util.eb.pointer import write_asm_pointer, from_snes_address, to_snes_address
-
+from coilsnake.util.eb.pointer import write_asm_pointer, write_xl_pointer, from_snes_address, to_snes_address
 
 class AsmPointer(object):
```
Comment (charasyn) [inline coilsnake/modules/eb/ExpandedTablesModule.py:7]: Can you:
1. Rename `AsmPointer` to `AsmPointerReference` (and similar with `XlPointer`)
2. Rename `write` to `relocate`
3. Move both classes to `coilsnake/util/eb/pointer.py`
4. Remove the subclassing of `object` (this is a Python2 leftover, and isn't necessary anymore)

Thanks!
P.S. If you have time and want to, go ahead and add a base class `PointerReference` with a `relocate` method which just raises `NotImplementedError` :)

**Review by charasyn — CHANGES_REQUESTED** [review]
I think we should move these Pointer classes to a different file and rename them.. Other than that the change looks good!

Just checking, this version shouldn't affect EbProjEdit compatibility unless the YML files actually have extra entries in them, right?

**Comment by tolmar** [comment]
> Just checking, this version shouldn't affect EbProjEdit compatibility unless the YML files actually have extra entries in them, right?

Yeah, EbProjEdit is fine until you actually use the new expansions.

**Review by charasyn — APPROVED** [review]
Looks good to me

**Pull request merged**